RocketChat/Rocket.Chat.ReactNative
# Chore: Migrate NewMessageView to Typescript

## Proposed changes

Migrate NewMessageView to Typescript and edit a prop as optional

## Issue(s)	


## How to test or reproduce

- When you are at Rooms List and click on the `Create` button, you should be moved to `NewMessageView`
- This is the screen migrated. You should test the search stuffs, creations differents rooms

## Screenshots

## Types of changes




## Checklist



## Further comments
